File: codechef/nov17/perpalin.py
t=int(input())
while t:
	n,p=map(int,input().split())
	if p==1 or p==2 or n==1 or n==2:
		print ("impossible")
	else:
		ans='a'+'b'*(p-2)+'a'
		for i in range(n//p):
			print (ans,end='')
		print()
	t-=1

File: codechef/nov17/perpalin.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
	int t;
	cin>>t;
	while(t--){
		long int n,p;
		cin>>n>>p;

		if(p==1 || p==2 || n==1 || n==2){
			cout<<"impossible"<<endl;
			continue;
		}

		string ans='a'+string(p-2,'b')+'a';
		for (long int i=0;i<n/p;i++){
			cout<<ans;
		}
		cout<<endl;
	}
	return 0;
	
}